### ISSUE TITLE
[Feature Request] Show all currently assigned Keyboard Shortcuts with commands as list

### ISSUE THREAD

**Issue by FreeCAD-Bug-Importer** [FreeCAD/FreeCAD#5596]
Issue imported from https://tracker.freecad.org/view.php?id=1756

* **Reporter:** r-frank
* **Date submitted:** 9/18/2014
* **FreeCAD version:** 0.14
* **Category:** Feature
* **Status:** assigned
* **Tags:** #lowhangingfruit, documentation, keyboard, shortcuts

# Original report text

Having a command (perhaps in the Tools-Section ?) which displays in a two cloumn list command and currently assigned keyboard shortcut. Maybe with possiblity to copy the list to the clipboard for inserting in Calc or Writer and then print it.

# Additional information

trimble Sketchup for example has a small ruby script that reads out this informations and writes the two column list in a HTML-File. This would be acceptable/conveniant too.

Forum thread: https://forum.freecadweb.org/viewtopic.php?f=8&t=23692

# Other bug information

* **Priority:** normal
* **Severity:** minor
* **Category:** Feature
* **Updated:** 2/6/2021

# Discussion from Mantis ticket

### Comment by Kunda1 2017-08-02 14:21
per triplus in https://forum.freecadweb.org/viewtopic.php?f=8&t=23692&p=184937#p184933
[quote=triplus]
When the time will allow and after a few other FreeCAD related projects will be finished ShortCuts module for FreeCAD will likely gain such functionality. It's on the todo list.

P.S. There is already such functionality in ShortCuts module. But it isn't yet extended to take "low-level" FreeCAD shortcuts into account.[/quote]

---

### Comment by Kunda1 2017-08-02 16:04
Opened issue at triplus's repo: https://github.com/triplus/ShortCuts/issues/1

---

### Comment by Kunda1 2018-03-07 21:21
Some good ideas here:
Dropdown menu that opens a PDF file:  https://youtu.be/PNZJox8pkls?t=5s

Find action dialog: https://youtu.be/PNZJox8pkls?t=17s

Keymap Preferences menu: https://youtu.be/PNZJox8pkls?t=45s

'Key Promoter' Plugin https://youtu.be/PNZJox8pkls?t=1m10s
(prompts you everytime you use the mouse, what keyboard shortcut you could of used instead)

**Comment by adrianinsaval** [comment]
this would be covered by #6506 once/if it gets merged I think

**Comment by maxwxyz** [comment]
Did https://github.com/FreeCAD/FreeCAD/pull/7771 solve this issue, as #6506 was replaced by this?

**Comment by wwmayer** [comment]
> Did 7771 solve this issue, as 6506 was replaced by this?

No, this issue is not solved. The reporter wanted a simple overview of all used shortcuts. Currently, you have to open the Customize dialog and navigate through all categories to see for which commands a shortcut is assigned. But this is very inconvenient.
